lib/ClusterProvider.py

```diff
@@ -0,0 +1,57 @@
+import common
+import subprocess
+import os
+import configs
+from robot.api import logger
+
+ROBOT_LIBRARY_SCOPE = 'SUITE'
+
+
+def load_script(provider):
+    path = 'scripts/cluster_providers/'
+    if provider == 'kube':
+        return path + 'kube.sh'
+    if provider == 'kind':
+        return path + 'kind.sh'
+    return "invalid"
+
+
+def setup_cluster(provider, metadata):
+    return SetupCluster(provider, metadata)
+
+
+class SetupCluster():
+    def __init__(self, provider, metadata):
+        self.provider = provider
+        self.metadata = metadata
+        self.base_cmd = 'bash -c'
+        self.cluster_name = 'helm-acceptance-test-' + self.metadata
+
+    def setup_cluster(self):
+        self.call_cluster_provisioner_script('create_cluster')
+        configs.AUTH_COMMAND = self.call_cluster_provisioner_script(
+            'get_cluster_auth')
```

Comment:
we are storing this for the kubeconfig with auth and cluster details right? 
we are getting the auth here but not returning anything. its seems redundant to me with the get_cluster_auth function

Reply:
Reason we have this because cluster provider class initialisation is happening in `kubernetes.robot` test, access to that instance isn't accessible to the all other `py` files unless we pass this instance explicitly to all needed py files. 
1. `kubernetes.py` -> initialise `ClusterProvider` instance based on provided variables.
2. In `ClusterProvider.py` -> `setup_cluster()` sets auth information related to this cluster in `configs.AUTH_COMMAND`, which will be then accessible to all other py files without needing to pass created `ClusterProvider` instance.

Reply:
yes, that's exactly my point was to keep the config file. but then in the `get_cluster_auth` function, do we want to retrieve the auth from cluster? That's why I was saying both the function are setting the config variable. only this function is not returning them.

Reply:
Fixed it now `get_cluster_auth` returns output string